### PR TITLE
feat: add `pr_branch_prefix` option to configure branch prefix for release-pr (#1728)

### PR DESCRIPTION
**Pull request by rajeshn76**
Fix for Issue # 1728  - Changes related to configuring the PR Branch Prefix in release-plz.toml (defaults to the current prefix "release-plz-" if none specified)

### CODE REVIEW

File: crates/release_plz/src/main.rs [hunk 1]
```diff
@@ -38,12 +38,14 @@ async fn run(args: CliArgs) -> anyhow::Result<()> {
...
             let config = cmd_args.update.config()?;
+            let pr_branch_prefix = config.workspace.pr_branch_prefix.clone();
             let pr_labels = config.workspace.pr_labels.clone();
             let pr_draft = config.workspace.pr_draft;
             let update_request = cmd_args.update.update_request(config, cargo_metadata)?;
             let request = ReleasePrRequest::new(update_request)
                 .mark_as_draft(pr_draft)
-                .with_labels(pr_labels);
+                .with_labels(pr_labels)
+                .with_branch_prefix(pr_branch_prefix);
```
Comment (MarcoIeni) [inline crates/release_plz/src/main.rs:48]: can you extract this `let request = ..` into a function?

something like `let request = get_release_pr_req(config)`;

File: crates/release_plz_core/src/command/release.rs [hunk 2]
```diff
@@ -65,6 +67,7 @@ impl ReleaseRequest {
             packages_config: PackagesConfig::default(),
             publish_timeout: minutes_30,
             release_always: true,
+            branch_prefix: BRANCH_PREFIX.to_string(),
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/command/release.rs:70]: can you rename `BRANCH_PREFIX` to `DEFAULT_BRANCH_PREFIX`?

File: crates/release_plz_core/src/command/release_pr.rs [hunk 3]
```diff
@@ -210,6 +235,8 @@ async fn create_pr(git_client: &GitClient, repo: &Repo, pr: &Pr) -> anyhow::Resu
     } else {
         create_release_branch(repo, &pr.branch)?;
     }
+    info!("changes committed to release branch {}", pr.branch);
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/command/release_pr.rs:238]:
```suggestion
    debug!("changes committed to release branch {}", pr.branch);
```

File: crates/release_plz_core/src/command/release_pr.rs [hunk 4]
```diff
@@ -274,10 +303,15 @@ fn update_pr_branch(
...
+fn reset_branch(
+    pr: &GitPr,
+    commits_number: usize,
+    repository: &Repo,
+    branch_prefix: &str,
+) -> anyhow::Result<()> {
     // sanity check to avoid doing bad things on non-release-plz branches
     anyhow::ensure!(
-        pr.branch().starts_with(BRANCH_PREFIX) || pr.branch().starts_with(OLD_BRANCH_PREFIX),
+        pr.branch().starts_with(branch_prefix) || pr.branch().starts_with(OLD_BRANCH_PREFIX),
```
Comment (MarcoIeni) [inline crates/release_plz_core/src/command/release_pr.rs:314]: let's add here `|| pr.branch().starts_with(BRANCH_PREFIX)`.
Not sure if it's needed but just to be safe.

**Comment by MarcoIeni** [comment]
Hey there! 👋

Thank you for your contribution. 🙏

Looks great, apart from a few comments I left 🙌

Can you also update the [docs](https://release-plz.ieni.dev/docs/config)?
In the docs, add a warning saying that before changing the release-plz branch you should close the old release PR.

File: crates/release_plz/src/main.rs (hunk 1 above)
Comment (rajeshn76) [reply crates/release_plz/src/main.rs]: extracted into a new function `get_release_pr_req`

File: crates/release_plz_core/src/command/release.rs (hunk 2 above)
Comment (rajeshn76) [reply crates/release_plz_core/src/command/release.rs]: renamed `BRANCH_PREFIX` to `DEFAULT_BRANCH_PREFIX`

File: crates/release_plz_core/src/command/release_pr.rs (hunk 3 above)
Comment (rajeshn76) [reply crates/release_plz_core/src/command/release_pr.rs]: Done

File: crates/release_plz_core/src/command/release_pr.rs (hunk 4 above)
Comment (rajeshn76) [reply crates/release_plz_core/src/command/release_pr.rs]: Added the extra check, just to be safe!

**Comment by rajeshn76** [comment]
Fixed all the code review comments and updated the config doc. Kindly review these changes.

**Review by MarcoIeni — APPROVED** [review]

**Pull request merged**